### PR TITLE
f-checkout@0.89.0 - Automatic scrolling to inline errors.

### PR DESCRIPTION
**Pull request by SachaZvetelman**
v0.89.0
------------------------------
*April 13, 2021*

### Added
- Automatic scrolling to the first inline error when trying to proceed to the payment page.

### CODE REVIEW

**Review by simonsandersje — DISMISSED** [review]

File: packages/components/organisms/f-checkout/src/components/Address.vue [hunk 1]
```diff
@@ -43,6 +44,7 @@
             <template #error>
                 <error-message
                     v-if="isAddressLocalityEmpty"
+                    data-error-message
```
Comment (kevinrodrigues) [inline packages/components/organisms/f-checkout/src/components/Address.vue]: We should probably hook the `js` name to this name, just to make it clear as to the reason it's here (a hook for javascript functionality)

`data-js-error-message`

Comment (SachaZvetelman) [reply packages/components/organisms/f-checkout/src/components/Address.vue]: I didn't know we had that naming convention, but sounds good to me. Will amend the code 👍

**Review by kevinrodrigues — DISMISSED** [review]

**Review by 1itvinka — DISMISSED** [review]

**Review by JoshuaNg2332 — DISMISSED** [review]

File: packages/components/organisms/f-checkout/src/components/Checkout.vue [hunk 2]
```diff
@@ -735,22 +736,18 @@ export default {
...
-                this.scrollToElement('errorAlert');
+                this.scrollToElement(this.$refs.errorAlert.$el);
             });
         },
 
         /**
          * Scroll to a ref element in the screen.
         */
-        scrollToElement (refElement) {
+        scrollToElement (element, options = { offset: -20 }) {
```
Comment (ashleynolan) [inline packages/components/organisms/f-checkout/src/components/Checkout.vue:746]: Small thing – would be good to update the comment for this function to describe the function params. It's not obvious why the default of offset would be -20 for instance, whereas a comment would describe that.

File: packages/components/organisms/f-checkout/src/components/Checkout.vue [hunk 3]
```diff
@@ -798,6 +797,17 @@ export default {
...
         },
 
+        /**
+         * Scroll to the first inline error, no matter which one it is.
+         */
+        scrollToFirstInlineError () {
+            this.$nextTick(() => {
+                const firstInlineError = document.querySelector('[data-js-error-message]');
+
+                this.scrollToElement(firstInlineError, { offset: -100 });
```
Comment (ashleynolan) [inline packages/components/organisms/f-checkout/src/components/Checkout.vue:807]: Probably would be cleared up by adding a comment to the `scrollToElement` function, but it's not obvious what -100 refers to (is it -100 pixels? Why 100 specifically – is that the height of an error element?)

**Pull request merged**